lnwallet/btcwallet/signer.go

```diff
@@ -370,21 +370,23 @@ func (b *BtcWallet) SignOutputRaw(tx *wire.MsgTx,
 		// slightly different, so we need to account for that to get the
 		// raw signature.
 		var rawSig []byte
-		if signDesc.TaprootKeySpend {
+		switch signDesc.SignMethod {
+		case input.SegWitV1KeySpendBIP0086SignMethod,
+			input.SegWitV1KeySpendRootHashSignMethod:
+
 			// This function tweaks the private key using the tap
-			// root key supplied as the tweak. So we pass in the
-			// original private key to avoid it being double
-			// tweaked!
+			// root key supplied as the tweak.
 			rawSig, err = txscript.RawTxInTaprootSignature(
 				tx, sigHashes, signDesc.InputIndex,
 				signDesc.Output.Value, signDesc.Output.PkScript,
-				signDesc.WitnessScript, signDesc.HashType,
+				signDesc.TapTweak, signDesc.HashType,
```

Comment:
IIRC my earlier diff had something here to force the value to be a `[]byte{}` for the bip86 case, is that no longer needed?

Reply:
I don't think that is necessary, no. Because nil and empty slice end up not being written to the hash anyway, it doesn't matter. It only used to matter when we used the field to _detect_ what signing method to use.

All those lines produce the same hash:
```
	t.Logf("Hash: %v", chainhash.TaggedHash(chainhash.TagTapTweak, []byte{1, 2, 3}))
	t.Logf("Hash: %v", chainhash.TaggedHash(chainhash.TagTapTweak, []byte{1, 2, 3}, []byte{}, []byte{}, nil))
	t.Logf("Hash: %v", chainhash.TaggedHash(chainhash.TagTapTweak, []byte{1, 2, 3}, []byte{}))
	t.Logf("Hash: %v", chainhash.TaggedHash(chainhash.TagTapTweak, []byte{1, 2, 3}, nil))
	t.Logf("Hash: %v", chainhash.TaggedHash(chainhash.TagTapTweak, []byte{1, 2, 3}, nil, nil, nil))
```